### DISCUSSION TITLE
pnpm with nvm on windows

### DISCUSSION THREAD

**Discussion by userquin** [pnpm/pnpm#4025] [Q&A] [answered]
It is a problem I have, we don't need to fix it on `pnpm`, but please don't change the name/extension again:

In some release of `pnpm` there was a small change on `pnpm.cmd`, it becomes to be `pnpm.CMD` (also applies to `pnpx.cmd`).

When switching between node versions via `nvm` on windows, the change keeps the `pnpm.[cmd|CMD]` from the original: at some point the link keeps the last switch.

For example, I have `node 12.18.3` and `node 16.13.0`: when switching between them the last one keeps on `node`: it also changes the link on the original.

**Comment by zkochan** [comment, marked as answer]
We won't change it again. It was a fix in pnpm [v6.16.0](https://github.com/pnpm/pnpm/releases/tag/v6.16.0). pnpm now reads the correct file extension from the PATHEXT env variable and uses it. In the past, pnpm was always using ".cmd", which was causing issues in case-sensitive directories on Windows.

**Reply by userquin** [reply]
thx @zkochan I love `pnpm`
